frontend/src/views/HomeView.vue

```diff
@@ -1,72 +1,51 @@
-<script lang="ts" setup>
-import { onMounted, ref } from 'vue'
-import { ElButton, ElDialog, ElInput, ElMessage } from 'element-plus'
+<template>
+  <div>
+    <el-button @click="showDialog">Add Knowledge</el-button>
+    <el-dialog v-model="dialogVisible" title="Add Knowledge">
+      <el-input v-model="newKnowledge" placeholder="Enter knowledge name"></el-input>
+      <span slot="footer" class="dialog-footer">
+        <el-button @click="dialogVisible = false">Cancel</el-button>
```

Comment:
Avoid to put logic inside template tag, always inside script tag. 
Can you put this click event in a function inside script tag?